src/macaron/slsa_analyzer/package_registry/maven_central_registry.py

```diff
@@ -236,3 +239,84 @@ def find_publish_timestamp(self, purl: str) -> datetime:
                 raise InvalidHTTPResponseError(f"The timestamp returned by {url} is invalid") from error
 
         raise InvalidHTTPResponseError(f"Invalid response from Maven central for {url}.")
+
+    @staticmethod
+    def get_artifact_file_name(purl: PackageURL) -> str | None:
```

Comment:
I think it's better if we rename this function and update the doc string to explicitly communicate that we only care about the "main" Maven artifact corresponding to a Maven type PackageURL. This is because for a GAV directory on Central repository for example, there can be many type of artifacts with different extensions and classifiers.
In additions, because the "main" Maven artifact doesn't have to always ends with ".jar". For example - https://repo1.maven.org/maven2/org/apache/batchee/batchee-webapp/1.0.3/, the "main" artifacts ends with ".war" instead. I wonder if we are going to support this case in the future ?

Reply:
I think it would make sense to specify the `jar` artefact as our actual target, currently.